Coding/fw/lib/WebServer/CaptivePortal.cpp

```diff
@@ -0,0 +1,171 @@
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Webserver
+@{
+@file       CaptivePortal.cpp
+
+Captive Portal for ESP32 WebServer @ref CaptivePortal.h
+
+* @}
+***************************************************************************************************/
+/* INCLUDES ***************************************************************************************/
+#include "CaptivePortal.h"
+#include "Web_Config.h"
+#include <SPIFFS.h>
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* CONSTANTS **************************************************************************************/
+
+/* MACROS *****************************************************************************************/
+
+/* TYPES ******************************************************************************************/
+static bool restartRequested = false;                 /**<  Variable to call Restart */
+/**************************************************************************************************/
+/**
+*  Processor to save the required credentials incoming from the Webpage
+* @param name Name of the parameter
+* @param value Value of the parameter 
+*/
+static void credentialsProcessor(String name, String value);
+
+/**
+* Captive portal request handler.
+*/
+class CaptiveRequestHandler : public AsyncWebHandler
+{
+public:
+    /**
+    * Constructs the captive portal request handler.
+    */
+    CaptiveRequestHandler()
+    {
+    }
+
+    /**
+    * Destroys the captive portal request handler.
+    */
+    ~CaptiveRequestHandler()
+    {
+    }
+
+    /**
+    * Checks whether the request can be handled.
+    * @param[in] request   Web request
+    * @return If request can be handled, it will return true otherwise false.
+    */
+    bool canHandle(AsyncWebServerRequest *request) override
+    {
+        /* The captive portal handles every request. */
+        return true;
+    }
+
+    /**
+    * Handles the request.
+    * @param[in] request   Web request, which to handle.
+    */
+    void handleRequest(AsyncWebServerRequest *request) override
+    {
+        if (nullptr == request)
+        {
+            return;
+        }
+        /** Forces authentication */
+        if (!request->authenticate(WebConfig::getWEB_USER().c_str(), WebConfig::getWEB_PASS().c_str()))
+        {
+            return request->requestAuthentication();
+        }
+
+        if (HTTP_POST == request->method())
+        {
+            if (request->args() != 0)
+            {
+                int params = request->params();
+                for (int i = 0; i < params; i++)
+                {
+                    AsyncWebParameter *p = request->getParam(i);
+                    credentialsProcessor(p->name(), p->value());
```

Comment:
does getParam always work ?   Return into p is not tested for null

Reply:
It is tested for Null in the getParam(i) function of the AsyncWebServer Lib (Line 92)